# ReadMe
1. Data Preparation
- Import 'master_data.csv' which has been prepared in 'COMP4135_Data_Preparation.ipynb'

2. Multiple Regression
- Use four independent variables at once to predict a dependent variable <br>

3. Independent Variable
- Average Watching Count 
- Video Completion Percentage 
- Enrolled Course Count 
- Average Enroll Time

4. Dependent Variable
- Grade

## Multiple Regression 

In [21]:
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt

In [22]:
#1 Import dataset
master_data = pd.read_csv('master_data.csv')
master_data.drop('Unnamed: 0', axis=1, inplace=True)
master_data

user_id  avg_watching_count  avg_completion_percentage  \
0      U_8126464           10.759358                   1.174700   
1      U_8650752            4.723404                   0.574468   
2       U_131074            2.555556                   0.000000   
3       U_262145            2.000000                   0.581395   
4      U_1441801            3.470588                   0.000000   
...          ...                 ...                        ...   
48635  U_2621421            6.419355                   0.274194   
48636  U_6684659            6.000000                   0.818182   
48637  U_8126455            1.358025                   0.777778   
48638  U_7995385            4.500000                   0.157895   
48639  U_6553596            5.800000                   0.616667   

       courses_count  avg_enroll_time  grade  
0                  5               21  76.27  
1                  5                9  61.35  
2                  6               17  69.09  
3                  4               15  67.20  
4                  5               16  68.02  
...              ...              ...    ...  
48635              8                7  59.61  
48636              4               18  71.51  
48637              9               11  63.77  
48638              4               22  74.77  
48639              4               15  67.97  

[48640 rows x 6 columns]

In [23]:
test_data = master_data.tail(1)
test_data

user_id  avg_watching_count  avg_completion_percentage  \
48639  U_6553596                 5.8                   0.616667   

       courses_count  avg_enroll_time  grade  
48639              4               15  67.97

In [24]:
master_data = master_data.head(48640-1)
master_data.shape

(48639, 6)

In [25]:
#2 Make a list of the independent values (X) and dependent values in a variable (Y).

# X = four features
# y = grade

X = master_data.iloc[:, 1:5].values
y = master_data.iloc[:, 5].values

In [26]:
#3 Splitting the dataset into the Training and Test dataset

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

In [27]:
#4 Create a multiple Linear Regression object

from sklearn.linear_model import LinearRegression
multiregr = LinearRegression()

In [28]:
#5 Fit the linear regression model to the training set
multiregr.fit(X_train, y_train)

LinearRegression()

In [29]:
#6 Predict the Test set results

y_pred= multiregr.predict(X_test)
y_pred

array([71.1422986 , 67.82824978, 75.46300876, ..., 66.73879963,
       73.49010605, 76.61782951])

In [30]:
y_test

array([71.14, 67.83, 75.46, ..., 66.74, 73.49, 76.62])

In [31]:
# correlation of 4 features
print("Correlation of four features = {}".format(multiregr.coef_))

Correlation of four features = [0.19077899 1.2150522  0.15923988 1.08695566]


<b>Interpretation of Correlation</b>
<li> One more time of watching video: 0.19077984 INCREASE in grade
<li> One more percent of completing video: 1.21503062 INCREASE in grade
<li> One more registered course: 0.15923719 INCREASE in grade
<li> One hour late for enrolling courses: 1.08695541 INCREASE in grade

## Demonstration

In [33]:
test_data

user_id  avg_watching_count  avg_completion_percentage  \
48639  U_6553596                 5.8                   0.616667   

       courses_count  avg_enroll_time  grade  
48639              4               15  67.97

In [32]:
# predict student's grade whose
    # average watching count = 20
    # average completion percentage = 0.5 (50%)
    # registered courses = 2
    # average enroll time = 23 (11pm)

predictedGrade = multiregr.predict([[5.8, 0.617, 4, 15]])
predictedGrade

array([64.49384974])

The actual grade is 67.97, and the result of multi-regression model is 64.49.

## Check Accuracy

In [43]:
from sklearn.linear_model import LinearRegression

In [45]:
r2_score = multiregr.score(X_test, y_test)
print(r2_score*100,'%')

99.99998105923528 %
